core-java-modules/core-java-io-5/src/test/java/com/baeldung/outputstreamtobytearray/OutputStreamToByteArrayUnitTest.java

```diff
@@ -0,0 +1,68 @@
+package com.baeldung.outputstreamtobytearray;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+import static org.junit.Assert.assertEquals;
+
+public class OutputStreamToByteArrayUnitTest {
+
+    @Test
+    public void givenFileOutputStream_whenUsingFileUtilsToReadTheFile_thenReturnByteArray(@TempDir Path tempDir) throws IOException {
+        String data = "Welcome to Baeldung!";
+        String fileName = "file.txt";
+        Path filePath = tempDir.resolve(fileName);
+
+        try (FileOutputStream outputStream = new FileOutputStream(filePath.toFile())) {
+            outputStream.write(data.getBytes(StandardCharsets.UTF_8));
+        }
+
+        byte[] writtenData = FileUtils.readFileToByteArray(filePath.toFile());
+        String result = new String(writtenData, StandardCharsets.UTF_8);
+        assertEquals(data, result);
+    }
+
+
+    @Test
+    public void givenFileOutputStream_whenUsingDrainableOutputStream_thenReturnByteArray(@TempDir Path tempDir) throws IOException {
+        String data = "Welcome to Baeldung!";
+        String fileName = "file.txt";
+        Path filePath = tempDir.resolve(fileName);
+
+        DrainableOutputStream drainableOutputStream = new DrainableOutputStream(new FileOutputStream(filePath.toFile()));
+
+        try {
+            drainableOutputStream.write(data.getBytes(StandardCharsets.UTF_8));
+        } finally {
+            drainableOutputStream.close();
+        }

```

Comment:
Use try-with-resources:
```suggestion
        DrainableOutputStream drainableOutputStream = new DrainableOutputStream(new FileOutputStream(filePath.toFile()));
        try (drainableOutputStream) {
            drainableOutputStream.write(data.getBytes(StandardCharsets.UTF_8));
        }
```